src/NLog/Targets/WebServiceTarget.cs

```diff
@@ -122,7 +123,20 @@ public WebServiceTarget(string name) : this()
         /// Gets or sets the web service URL.
         /// </summary>
         /// <docgen category='Web Service Options' order='10' />
-        public Uri Url { get; set; }
+        /// 
+        public Layout Url { get; set; }
+        internal Uri _renderedUrl
+        {
+            get
+            {
+                var uri = RenderLogEvent(Url, LogEventInfo.CreateNullEvent());
```

Comment:
No need to render every time when using `LogEventInfo.CreateNullEvent()`. Instead cache the rendering, and reset the cache in InitializeTarget.